src/plugins/posix/linux_aio_io_queue.cpp

```
@@ -122,75 +148,185 @@ nixlPosixIOQueueLinuxAIO::post(void) {
-        completed_ios.push_back(io);
+    for (auto &io : ios_) {
+        if (io.in_flight_) {
+            failIO(&io);
+        }
     }
+}
 
-    if (!completed_ios.empty()) {
-        free_ios_.splice(free_ios_.end(), completed_ios);
+unsigned
+nixlPosixIOQueueLinuxAIO::cancel(void *ctx, nixlPosixIOQueueCancelDoneCb) {
```

Cppcheck (2.21.0)

[style] The function 'cancel' is never used.

(unusedFunction)